jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/ProgressDisplayGenerator.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.plugins.common;
+
+import com.google.cloud.tools.jib.event.progress.Allocation;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Generates a display of progress and unfinished tasks.
+ *
+ * <p>Example:
+ *
+ * <p>Executing tasks... [================= ] 72.5% complete > task 1 running > task 3 running
```

Comment:
The HEADER, the bar, and the two tasks appear in a single line here. (Not that it cannot be.)

Reply:
Oh oops the formatting got messed up